### PR TITLE
Add batch file reader

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFileP.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    private void processFile(Path file) {
+        if (getLogger().isFinestEnabled()) {
+            getLogger().finest("Processing file " + file);
+        }
+
+        try (
+            BufferedReader reader = Files.newBufferedReader(file, charset)
+        ) {
+            String line;
+            while ((line = reader.readLine()) != null) {
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFileP.java]: I would prefer `for (String line; line = reader.readLine() != null; )` here

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFileP.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        return ((hashCode & Integer.MAX_VALUE) % parallelism) == id;
+    }
+
+    private void processFile(Path file) {
+        if (getLogger().isFinestEnabled()) {
+            getLogger().finest("Processing file " + file);
+        }
+
+        try (
+            BufferedReader reader = Files.newBufferedReader(file, charset)
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFileP.java]: can we collapse this into one line? try (BufferedReader.. )

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFileP.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        this.charset = charset;
+        this.parallelism = parallelism;
+        this.id = id;
+    }
+
+    @Override
+    public boolean complete() {
+        DirectoryStream<Path> directoryStream;
+        try {
+            directoryStream = Files.newDirectoryStream(directory, glob == null ? "*" : glob);
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFileP.java]: should be in a try() {} block.

Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFileP.java]: I didn't know, that DirectoryStream needs to be closed...

**Comment by viliam-durina** [comment]
verify

**Comment by viliam-durina** [comment]
verify

**Pull request merged**